specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -0,0 +1,226 @@
+	encodeAddress(publicKey: Buffer): string {
+		const keyHash: string = sha512_256.create().update(publicKey).hex()
+
+		// last 4 bytes of the hash
+		const checksum: string = keyHash.slice(-8)
+
+		return base32.encode(Encoder.ConcatArrays(publicKey, Buffer.from(checksum, "hex"))).slice(0, 58)
+	}
+```
+
+
+### Encoding
+
+#### Types
+
+Before encoding, all `txn` parameters **MUST** be converted into primitive types:
+
+- `gh`: The genesis hash of the network, which is a base64 encoded string should be decoded. 
+- `snd`: The sender address should be decoded into its public key bytes.
+- `rcv`: The receiver address should be decoded into its public key bytes.
+- `xaid`: The asset ID should be converted into a uint64 integer.
+- `aamnt`: The amount to transfer should be converted into a uint64 integer.
+- `fee`: The fee for the transaction should be converted into a uint64 integer.
+- `fv`: The first valid round should be converted into a uint64 integer.
+- `lv`: The last valid round should be converted into a uint64 integer.
+
+#### Scheme
+
+Algorand uses [**msgpack**](https://www.npmjs.com/package/algorand-msgpack) to encode the transaction before signing.
+
+
+#### Steps to sign
+
+- Decode the `txn` parameters as described above.
+- Sort fields alphabetically. 
+- Encode the sorted fields using msgpack.
+- Add a prefix `TX` in its byte representation to the encoded message.
+- Sign the resulting byte array using the sender's private key with the `Ed25519` signature scheme.
+
+
+## Full `X-PAYMENT` header:
+
+```json
+{
+  "x402Version": 1,
+  "scheme": "exact",
+  "network": "algorand-mainnet",
+  "payload": {
+    "paymentTxID": "R4PFEBJGLXV4PDOIBLY7UKBLVLXFCPYLXGBILC22BAVILTKWV3TA", // ID of the transaction in the group that will pay the resource server
+    "paymentGroup": [
+      {
+        "signature": "VxpW6emWwct7ymzyXa/F2T7rwepM71XwsslPKEdpVqmbcRUB28QMdYJuuLkeP4dnVoI7emmxT6Q/YktqHc8xBg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "fee": "2000",
+          "fv": "6631154",
+          "lv": "6632154",
+          "snd": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "rcv": "LOXIQKHRCEDEMN5MKBQ32Y54J7H6JKBTEURQL4S65K44MTWN6UMTU2WNBM",
+          "amnt": "0",
+          "type": "pay"
+        }
+      },
+      {
+        "signature": "wVGxbZwPd2c8zKX4RSE4muFRBOHe/nRq5VT6iMeMksi2agfOCTs+VzOGUPJ+bTT5T21EuYjoBKKDX2+/NZG8Cg==",
+        "txn": {
+          "gen": "mainnet-v1.0",
+          "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
+          "asnd": "PPNHVQJGE6ZMEWPR35UHLUYMCCZV6VNTHLJMZDVCONXKUPV47K4WGQJWGI",
+          "arcv": "ACTSMNPJPS5JMZJJ5G73JOXUUMWXXDGS7TMOER3M4W7BC54ERSZ4KPSMHM",
+          "aamt": "1000000",
+          "type": "axfer",
+          "fee": "0",
+          "fv": "6631154",
+          "lv": "6632154",
+          "xaid": "31566704"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Verification
```

Comment:
This process does not explain how to find the `paymentTxID` in the group (the `TxID` must be computed).

Reply:
The `PaymentTxID` is supplied as part of the header.

Reply:
Added explanation how to calculate it